tests/unit/src/test/scala/tests/BatchedFunctionSuite.scala

```diff
@@ -36,4 +36,39 @@ object BatchedFunctionSuite extends BaseSuite {
     }
   }
 
+  testAsync("accumulate") {
+    val lock = new Object
+    val mkString = BatchedFunction.fromFuture[String, String]({ numbers =>
+      Future {
+        lock.synchronized {
+          numbers.mkString
+        }
+      }
+    })
+    val results = lock.synchronized {
+      mkString.accumulate
```

Comment:
Tried 
```scala
    val mkString = BatchedFunction.fromFuture[String, String]{ numbers =>
      Future.successful {
        numbers.mkString
      }
    }

    assertEquals(mkString("a").value.isDefined, true)
  }
```
but test fails. Probably we cannot make it fully synchronous because of use Promise inside the `BatchedFunction`

Reply:
@olafurpg do you have any idea how should I approach that test as this does not work as expected?

Reply:
Here's a fully synchronous test that passes on my local computer
```scala
  test("pause") {
    implicit val ec: ExecutionContext = new ExecutionContext {
      def execute(runnable: Runnable): Unit = runnable.run()
      def reportFailure(cause: Throwable): Unit = cause.printStackTrace()
    }
    val lock = new Object
    val mkString = BatchedFunction.fromFuture[String, String]({ numbers =>
      Future.successful {
        numbers.mkString
      }
    })
    val unpaused = mkString(List("a"))
    assert(unpaused.isCompleted)
    mkString.accumulate
    val paused = mkString("b")
    assert(!paused.isCompleted)
    val paused2 = mkString("c")
    assert(!paused2.isCompleted)
    mkString.restart
    assert(paused.value == Some(Success("bc")))
    assert(paused2.value == Some(Success("bc")))
  }
```

Reply:
Sorry, I didn't notice your comment. Done. Thanks for help!